File: Java/Others/LargestPositive.java
/*Given an integer array that does not contain any zeros, find the largest positive integer k such that -k also exists in the array. Return the positive integer k. If there is no such integer, return -1*/

import java.util.Scanner;

class LargestPositive {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter an Integer array, with space to seperate each element");
        String[] input = sc.nextLine().split("\\s+");

        int arraySize = input.length;
        int[] array = new int[arraySize];
        for(int i = 0; i < arraySize; i++) {
            array[i] = Integer.parseInt(input[i]);
        }
        
        int largestNumber = findLargest(array,arraySize);
        
        if(largestNumber>0) {
            System.out.println("Largest positive number in array that has its alternative negative number is: " +largestNumber);
        } else {
            System.out.println("No such element found");
        }
    }

    static int findLargest(int[] array, int arraySize)
    {
        int largest = -1;
        for(int i=0; i<arraySize; i++) {
            for(int j=i+1; j<arraySize; j++) {
                int sum = array[i]+array[j];
                if(sum==0) {
                    largest = Math.max(largest,Math.abs(array[i]));
                }
            }
        }
        return largest;
    }
}

File: Kotlin/Others/LargestPositive.kt
/*Given an integer array that does not contain any zeros, find the largest positive integer k such that -k also exists in the array. Return the positive integer k. If there is no such integer, return -1*/

import kotlin.math.*

fun main() {
  println("Enter an Integer array, with space to seperate each element")
  val input = readln().split("\\s+".toRegex())
  
  val arraySize = input.size
  var array = IntArray(arraySize)
  for(i in 0 until arraySize) {
      array[i] = input[i].toInt()
  }
  
  val largestNumber = findLargest(array,arraySize)
  
  if(largestNumber>0) {
    println("Largest positive number in array that has its alternative negative number is: $largestNumber")
  } else {
    println("No such element found")
  }
}

fun findLargest(array : IntArray, arraySize : Int) : Int {
  var largest = -1
  for(i in 0 until arraySize) {
    for(j in i+1 until arraySize) {
      var sum = array[i]+array[j]
        if(sum==0) {
          largest = max(largest,abs(array[i]))
        }
    }
  }
  return largest
}